src/satisfia/agents/makeMDPAgentSatisfia.py

```diff
@@ -0,0 +1,250 @@
+
+		# register (state, action) in global store (could be anywhere, but here is just as fine as anywhere else)
+		stateActionPairsSet.insert((state, action))
+
+		Edel = expectedDelta(state, action)
+		if state.terminateAfterAction:
+			q = Edel
+		else:
+			# Bellman equation
+			q = distribution.infer(lambda: self.minAdmissibleV(self.world.transition(state, action))).E() # recursion
+
+		if verbose or debug:
+			print("minAdmissibleQ, state", state, "action", action, ":", q)
+
+		return q
+
+	@lru_cache(maxsize=None)
+	def maxAdmissibleV(self, state): # recursive
+		if verbose or debug:
+			print("maxAdmissibleV, state", state, "...")
+
+		actions = self.world.stateToActions(state)
+		qs = [self.maxAdmissibleQ(state, a) for a in actions] # recursion
+		v = max(qs) if maxLambda == 1 else interpolate(min(qs), maxLambda, max(qs))
+
+		if verbose or debug:
+			print("maxAdmissibleV, state", state, ":", v)
+
+		return v
+
+	@lru_cache(maxsize=None)
+	def minAdmissibleV(self, state): # recursive
+		if verbose or debug:
+			print("minAdmissibleV, state", state, "...")
+
+		actions = self.world.stateToActions(state)
+		qs = [self.minAdmissibleQ(state, a) for a in actions] # recursion
+		v = min(qs) if minLambda == 0 else interpolate(min(qs), minLambda, max(qs))
+
+		if verbose or debug:
+			print("minAdmissibleV, state", state, ":", v)
+
+		return v
+
+	# The resulting admissibility interval for states.
+	def admissibility4state(self, state):
+		return self.minAdmissibleV(state), self.maxAdmissibleV(state)
+
+	# The resulting admissibility interval for actions.
+	def admissibility4action(self, state, action):
+		return self.minAdmissibleQ(state, action), self.maxAdmissibleQ(state, action)
+
+
+
+
+	# Some safety metrics do not depend on aspiration and can thus also be computed upfront,
+	# like min/maxAdmissibleQ, min/maxAdmissibleV:
+
+	# Messing potential (maximal entropy (relative to some uninformedStatePrior) 
+	# over trajectories any agent could produce from here (see overleaf for details)):
+	@lru_cache(maxsize=None)
+	def messingPotential_action(self, state, action): # recursive
+		def sample():
+			if state.terminateAfterAction:
+				return 0
+			else:
+				# TODO understand these two lines
+				nextState = self.world.transition(state, action)
+				nextStateScore = self.transitionDistribution(state, action).score(nextState)
+
+				nextMP = self.messingPotential_state(nextState) # recursion
+				priorScore = self["uninformedStatePriorScore"](state) if self["uninformedStatePriorScore"] else 0
+				internalEntropy = self["internalTransitionEntropy"](state, action, nextState) if self["internalTransitionEntropy"] else 0
+				return nextMP + priorScore - nextStateScore + internalEntropy;
+
+		# Note for ANN approximation: messingPotential_action can be positive or negative. 
+		return distribution.infer(sample).E()
+
+	@lru_cache(maxsize=None)
+	def messingPotential_state(self, state): # recursive
+		actions = self.world.stateToActions(state)
```

Comment:
I have added to WorldModel a function `possible_actions` (https://github.com/pik-gane/satisfia/blob/world_model/src/world_model/world_model.py#L90) that we can use here:
```
actions = self.world_model.possible_actions(state)
```